include/swift/SILOptimizer/Utils/PartitionUtils.h

```diff
@@ -118,14 +118,24 @@ class SILIsolationInfo {
 
   /// This is the value that we got isolation from if we were able to find
   /// one. Used for isolation history.
-  SILValue isolationSource;
+  SILValue isolatedValue;
 
-  SILIsolationInfo(ActorIsolation actorIsolation, SILValue isolationSource)
+  /// If set this is the SILValue that represents the actor instance that we
+  /// derived isolatedValue from.
+  SILValue actorInstance;
+
+  SILIsolationInfo(ActorIsolation actorIsolation, SILValue isolatedValue,
+                   SILValue actorInstance)
       : kind(Actor), actorIsolation(actorIsolation),
-        isolationSource(isolationSource) {}
+        isolatedValue(isolatedValue), actorInstance(actorInstance) {
+    assert((!actorInstance ||
+            (actorIsolation.getKind() == ActorIsolation::ActorInstance &&
+             actorInstance->getType().isActor())) &&
```

Comment:
```suggestion
             actorInstance->getType().isAnyActor())) &&
```

otherwise would crash on a distributed actor

Reply:
This actually calls:

```
bool TypeBase::isActorType() {                                                                                            
  if (auto actor = getAnyActor())                                                                                         
    return actor->isActor();                                                                                              
  return false;                                                                                                           
} 
```

which does any actor.

Reply:
Right, which is still wrong -- it does isActor but should do `isAnyActor` here